### PR TITLE
Add site health checks for Ads module.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #8245

## Relevant technical choices

In order to implement `ModuleWithTag` I had to implement a `register_tag` function:

https://github.com/google/site-kit-wp/blob/9a2dfa37f626b6ed9f96043a0b2b8cf4763bad58/includes/Modules/Ads.php#L132-L135

I also tested the regex used would capture the provided gtag config call.

This PR returns the Ads Conversion ID to the Site Health Checks but without the tag registered I was not able to see the tag registration check on this page as implemented.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Modules/Ads/Tag_Matchers.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	 * Holds array of regex tag matchers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Array of regex matchers.
+	 */
+	public function regex_matchers() {
+		return array(
+			// Detect Ads gtag config with valid AdWords Conversion ID.
+			"/gtag\(['|\"]config['|\"],['|\"]AW-[0-9]+['|\"]\)/",
```
Comment (benbowler) [inline includes/Modules/Ads/Tag_Matchers.php:35]:
```suggestion
			"/gtag\(['|\"]config['|\"],\s*['|\"]AW-[0-9]+['|\"]\)/",
```

**Review by zutigrm — APPROVED** [review]
Thanks @benbowler LGTM.

I synced this branch with develop since the tag implementation was needed to confirm the behaviour.

I also updated the QAB to describe all cases, and take new tag implementation into account, which now brings the inline comment used to detect that tag is placed by Site Kit.

Note to the Merge reviewer: There is a weird CI only test failure for php 5.6, it might be some glitch, couldn't replicate it locally, and the code change here is very small to affect it

File: includes/Modules/Ads.php [hunk 2]
```diff
@@ -33,8 +35,9 @@
  * @access private
  * @ignore
  */
-final class Ads extends Module implements Module_With_Assets, Module_With_Settings, Module_With_Tag {
+final class Ads extends Module implements Module_With_Assets, Module_With_Debug_Fields, Module_With_Settings, Module_With_Tag {
 	use Module_With_Assets_Trait;
+	use Module_With_Tag_Trait;
```
Comment (eugene-manuilov) [inline includes/Modules/Ads.php:40]: This is not needed because it is already added to the class.

File: includes/Modules/Ads.php [hunk 3]
```diff
@@ -159,6 +162,24 @@ public function register_tag() {
 		$tag->register();
 	}
 
+	/**
+	 * Gets an array of debug field definitions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
```
Comment (eugene-manuilov) [inline includes/Modules/Ads.php:170]: Let's add a description for what is returned from this function to make it consistent with other method phpdocs.

File: tests/phpunit/integration/Modules/AdsTest.php [hunk 4]
```diff
@@ -50,6 +51,19 @@ public function test_settings_reset_on_deactivation() {
 		$this->assertFalse( $ads_settings );
 	}
 
+	public function test_get_debug_fields() {
+		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+
+		$ads->get_settings()->set( array( 'adsConversionID' => 'AW-123456789' ) );
+
+		$this->assertEqualSets(
```
Comment (eugene-manuilov) [inline tests/phpunit/integration/Modules/AdsTest.php:59]: Let's also verify that the value is the same as we set in the settings.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @benbowler. A few comments from me. Please, take a look.

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**